src/clib/pio_darray.c

```diff
@@ -2874,7 +2881,8 @@ static int PIOc_read_darray_adios(file_desc_t *file, int fndims, io_desc_t *iode
                            pio_get_vname_from_file(file, varid), varid, pio_get_fname_from_file(file), file->pio_ncid);
         }
 
-        int32_t decomp_blocks_size = decomp_blocks->nblocks;
+        size_t decomp_blocks_size = decomp_blocks->nblocks;
```

Comment:
Any reason this variable was using a fixed sized int type before?

Reply:
I do not know why Dmitry used int32_t before. I think we should use size_t here to be consistent with ADIO2 lib.
FYI, in https://github.com/ornladios/ADIOS2/blob/master/bindings/C/adios2/c/adios2_c_types.h:
```
typedef struct
{
    int Dims;
    size_t *Shape;
    int IsValue;
    int IsReverseDims;
    size_t nblocks;
    adios2_blockinfo *BlocksInfo;
} adios2_varinfo;
```